File: app/src/main/java/com/example/smishingdetectionapp/SmishingDetectionApp.java
package com.example.smishingdetectionapp;
import android.app.Application;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import androidx.appcompat.app.AppCompatDelegate;

public class SmishingDetectionApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();

        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(this);
        boolean darkModeEnabled = prefs.getBoolean("dark_mode", false); // default = light

        AppCompatDelegate.setDefaultNightMode(
                darkModeEnabled
                        ? AppCompatDelegate.MODE_NIGHT_YES
                        : AppCompatDelegate.MODE_NIGHT_NO
        );
    }

}

File: app/src/main/java/com/example/smishingdetectionapp/SmishingDetectionApp.kt
package com.example.smishingdetectionapp

import android.app.Application
import android.preference.PreferenceManager
import androidx.appcompat.app.AppCompatDelegate

class SmishingDetectionApp : Application() {
    override fun onCreate() {
        super.onCreate()

        val prefs = PreferenceManager.getDefaultSharedPreferences(this)
        val darkModeEnabled = prefs.getBoolean("dark_mode", false) // default = light

        AppCompatDelegate.setDefaultNightMode(
            if (darkModeEnabled)
                AppCompatDelegate.MODE_NIGHT_YES
            else
                AppCompatDelegate.MODE_NIGHT_NO
        )
    }
}